src/ir/mod.rs

```diff
@@ -84,8 +87,11 @@ impl Environment {
     pub fn add_clauses<I>(&self, clauses: I) -> Arc<Environment>
         where I: IntoIterator<Item = DomainGoal>
     {
+        use std::collections::HashSet;
```

Comment:
I'm perfectly ok with moving it to the top. I don't know why but some times I prefer local imports if they're used only at one little place. Not every time though haha.